agent/metrics/generic_metrics_client.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
We will only enable metrics collection on Linux. Because the global variable is accessed throughout Agent, this code needs to be built. However, if not opted in, recordGenericMetric(...) in metrics_engine.go will do nothing.